src/js/views/forms/form/form_views.js

```diff
@@ -259,21 +260,94 @@ const ReadOnlyView = View.extend({
   template: hbs`{{ @intl.forms.form.formViews.readOnlyView.buttonText }}`,
 });
 
+const SaveButtonTypeDroplist = Droplist.extend({
+  popWidth: 192,
+  align: 'right',
+  initialize({ model }) {
+    this.collection = new Backbone.Collection([
+      {
+        text: i18n.saveView.save.droplistItemText,
+        value: 'save',
+      },
+      {
+        text: i18n.saveView.saveAndGoBack.droplistItemText,
+        value: 'saveAndGoBack',
+      },
+    ]);
+
+    const currentSaveButtonType = model.get('saveButtonType');
+
+    this.setState('selected', this.collection.find({ value: currentSaveButtonType }));
+  },
+  viewOptions: {
+    className: 'button--green form__save-dropdown-button',
+    template: hbs`{{fas "caret-down"}}`,
+  },
+  picklistOptions() {
+    return {
+      headingText: i18n.saveView.droplistLabel,
+      itemTemplate: hbs`
+        <div class="form__save-dropdown-picklist-item">
+          {{text}}
+          <span class="form__save-dropdown-picklist-icon">{{#if isSelected}}{{fas "circle-check"}}{{/if}}</span>
+        </div>
+      `,
+      itemTemplateContext() {
+        return {
+          isSelected: this.model === this.state.get('selected'),
+        };
+      },
+    };
+  },
+});
+
 const SaveView = View.extend({
-  isDisabled: false,
-  tagName: 'button',
-  className: 'button--green',
-  attributes() {
+  className: 'form__save-button-container',
+  regions: {
+    saveType: {
+      el: '[data-save-type-region]',
+      replaceElement: true,
+    },
+  },
+  modelEvents: {
+    'change:saveButtonType': 'render',
+  },
+  templateContext() {
+    const saveButtonType = this.model.get('saveButtonType');
+
     return {
-      disabled: this.getOption('isDisabled'),
+      isDisabled: this.getOption('isDisabled'),
+      showSaveButton: saveButtonType === 'save',
+      showSaveGoBackButton: saveButtonType === 'saveAndGoBack',
     };
   },
-  template: hbs`{{ @intl.forms.form.formViews.saveView.buttonText }}`,
+  template: hbs`
+    <button class="button--green form__save-button js-save-button" {{#if isDisabled}}disabled{{/if}}>
+      {{#if showSaveButton}}
+        {{ @intl.forms.form.formViews.saveView.save.buttonText }}
+      {{/if}}
+      {{#if showSaveGoBackButton}}
+        {{ @intl.forms.form.formViews.saveView.saveAndGoBack.buttonText }}
+      {{/if}}
+    </button>
+    <button data-save-type-region></button>
+  `,
   triggers: {
-    'click': 'click',
+    'click .js-save-button': 'click:save',
```

Comment:
Changed this from `click` to `click:save`. Thought it made more sense in terms of code readability, since there are now multiple clicks going on in this view. Not just the `Save` button. But the droplist button now too.